File: 53 - FizzBuzz V2/solution53_test.py
import unittest
from io import StringIO
import sys
from solution53 import fizzbuzz

class TestFizzBuzz(unittest.TestCase):
    def test_fizzbuzz(self):
        output = StringIO()
        sys.stdout = output

        fizzbuzz()

        sys.stdout = sys.__stdout__

        result = output.getvalue().strip().split('\n')

        expected = [
            '1', '2', 'fizz', '4', 'buzz', 'fizz', '7', '8', 'fizz', 'buzz',
            '11', 'fizz', '13', '14', 'fizzbuzz', '16', '17', 'fizz', '19', 'buzz'
        ]
        self.assertEqual(result[:20], expected)

if __name__ == '__main__':
    unittest.main()

File: 53 - FizzBuzz V2/solution53_test.go
package main

import (
	"testing"
)

func TestFizzBuzz(t *testing.T) {
	tests := []struct {
		input    int
		expected string
	}{
		{1, "1"},
		{2, "2"},
		{3, "fizz"},
		{5, "buzz"},
		{15, "fizzbuzz"},
		{30, "fizzbuzz"},
		{7, "7"},
	}

	for _, test := range tests {
		result := fizzBuzz(test.input)
		if result != test.expected {
			t.Errorf("fizzBuzz(%d) = %s; want %s", test.input, result, test.expected)
		}
	}
}
